packages/clerk-js/src/core/state.ts

```diff
@@ -0,0 +1,33 @@
+import type { State as StateInterface } from '@clerk/types';
+import { computed, effect } from 'alien-signals';
+
+import { eventBus } from './events';
+import type { BaseResource } from './resources/Base';
+import { SignIn } from './resources/SignIn';
+import { signInComputedSignal, signInErrorSignal, signInSignal } from './signals';
+
+export class State implements StateInterface {
```

Comment:
JSDoc would be good here 👀  what's it for? Maybe it belongs on the `State` type instead.